### DISCUSSION TITLE
Lidarr integration?

### DISCUSSION THREAD

**Discussion by rwjack** [music-assistant/support#2393] [Feature requests and ideas]
Any thoughts on having an option to search for and download a song via Lidarr, through Music Assistant?

**Comment by OzGav** [comment]
Is there any legal use case for this?

**Reply by rwjack** [reply]
![IMG_6799](https://github.com/music-assistant/hass-music-assistant/assets/59068073/4c4340db-1c0a-413f-a1ec-39118a3264c8)

**Reply by OzGav** [reply]
What is that supposed to be?

**Reply by rwjack** [reply]
But yeah, basically downloading music/playlists I listen to on Spotify(premium), so I can listen to them offline, and so I don't have a 3 second delay on Snapcast when playing/pausing/skipping media.

**Reply by OzGav** [reply]
We do not allow any form of downloading of music from a streaming provider.

**Reply by rwjack** [reply]
Of course, that would be illegal, hence the Lidarr integration.

**Reply by OzGav** [reply]
How is Lidarr permitting legal access to music you listen to on Spotify?

**Reply by rwjack** [reply]
Music assistant has access to my Spotify account, hence it collects information on what I listen to. The Music assistant - Lidarr integration could then be configured to automatically download songs based on this information, or there could be a little download button next to each song.

To clarify in legaleze, the songs are not being downloaded from Spotify, or any other streaming provider for that matter.

**Reply by OzGav** [reply]
Then I am confused as you say

> could then be configured to automatically download songs 

Are you suggesting it is ok to download the music from somewhere (eg. usenet or bittorrent) as long as it is not a streaming provider?

**Reply by rwjack** [reply]
Yes, I am suggesting so, since as I mentioned, saving and redistributing music from a provider such as Spotify, would be illegal.

**Reply by OzGav** [reply]
OK let me be clear then. This project will not allow, facilitate or condone music piracy in any form. If you suggest such a thing again you will be blocked.

**Reply by rwjack** [reply]
1. You seem to know a lot more about "usenet" and torrenting than me, yet you accuse me of condoning piracy, which I never mentioned directly.
2. Unfortunate, but oddly common to see developers of such a cool projects having awfully fragile egos, whipping out the ban hammer just like that. You could have just said, "No, we won't be doing Lidarr, period."
3. Refer to my GIF above.
4. Have a nice day.
